backend/installGlobalHook.js

```diff
@@ -141,14 +128,45 @@ function installGlobalHook(window: Object) {
     }
     return 'production';
   }
+
+  let hasDetectedBadDCE = false;
+
   const hook = ({
     // Shared between Stack and Fiber:
     _renderers: {},
     helpers: {},
+    checkDCE: function(fn) {
+      // This runs for production versions of React.
+      // Needs to be super safe.
+      try {
+        var toString = Function.prototype.toString;
+        var code = toString.call(fn);
+        // This is a string embedded in the passed function under DEV-only
+        // condition. However the function executes only in PROD. Therefore,
+        // if we see it, dead code elimination did not work.
+        if (code.indexOf('^_^') > -1) {
```

Comment:
Heh, should we put something slightly more descriptive? 😁
I guess this pretty unique and tiny which is nice. It's just so arbitrary.

**Edit** Looks like the corresponding change has already been merged to react so nevermind. :)

Reply:
I figured I didn't want it to be long, but also didn't want it to be something that could accidentally appear in that function. Since then it would cause false positive.